bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/ScanAndCompareGarbageCollector.java

```diff
@@ -153,6 +156,7 @@ public void gc(GarbageCleaner garbageCleaner) {
             Versioned<LedgerMetadata> metadata = null;
             while (!done) {
                 start = end + 1;
+                gcRateLimiter.tryAcquire();
```

Comment:
You are not checking the `tryAcquire()` result, and `gcRateLimiter.tryAcquire();` changed nothing, so it cannot block the ZK requests.

Reply:
You are right, thanks for pointing out this. I have updated the implementation.
PTAL again, thanks.